### PR TITLE
Fix usage of struct tags for config

### PR DESCRIPTION
**Pull request by mraerino**
#183 introduced a breaking change that meant the struct tag `mapstructure` could no longer be used to map config keys.

since this is used in various services we need to restore this behaviour.

### CODE REVIEW

File: nconf/configuration.go [hunk 1]
```diff
@@ -24,15 +35,26 @@ func LoadFromFile(configFile string, input interface{}) error {
...
+	case ".yaml", ".yml":
+		err = yaml.Unmarshal(data, &config)
+	default:
+		err = &ErrUnknownConfigFormat{configExt}
+	}
+	if err != nil {
+		return fmt.Errorf("failed to read config: %w", err)
+	}
+
+	if err := mapstructure.Decode(&config, &input); err != nil {
```
Comment (mheffner) [inline nconf/configuration.go:54]: Should this be just `input`?

Comment (mraerino) [reply nconf/configuration.go:54]: mapstructure wants a pointer to a map

Comment (mraerino) [reply nconf/configuration.go]: ah, yup. changed.

**Review by mheffner — APPROVED** [review]
:+1:

**Pull request merged**